### ISSUE TITLE
Feature requests: Directory playback, etc

### ISSUE THREAD

**Issue by LukeZBaker** [fooyin/fooyin#54]
Directory playback i.e. the ability to traverse a file system and play tracks/folders
Support for CUE sheets
Download cover art
Ability to export playlists

**Comment by ludouzi** [comment]
CUE support, downloading cover art and exporting play lists are already planned. 

I can see the appeal of a directory browser, and it would be fairly trivial to implement, so I'll add one after I've setup packaging.

**Comment by LukeZBaker** [comment]
Thanks, there are a few directory browser music players but none of the GUI ones are particularly awesome. The feature  would appeal to lots of people who don't have the inclination to fix their metadata. So it's great you are planning to add it.

Very happy that the other suggestions are already planned. It might be worth adding a ROADMAP so we don't suggest things that are already on your radar!

**Comment by ludouzi** [comment]
Agreed. I'm also thinking of a way to use a libraries structure for metadata for tracks without any tags i.e. Artist/Album/Track - Title. 

Good call on the roadmap. It'll also be handy in case of other contributors.

**Comment by LukeZBaker** [comment]
If you need any help testing this, or anything else to test, I'm happy to help.

I've had the odd segmentation fault with the current release.  If I can enable logs etc please let me know how, I did build with -DBUILD_WERROR but couldn't see any crash information.

**Comment by ludouzi** [comment]
Thanks, I appreciate the offer.

Log statements are printed to terminal currently, but for seg faults a backtrace is really needed from running in a debugger. For now, the best way to help in that regard is to let me know what you were doing just before the crash (if possible).

BUILD_WERROR is used only when building and it replaces warnings (which still allow the program to be compiled) with errors (which stop the build).

**Comment by LukeZBaker** [comment]
Thanks for adding the directory browser. 

If there's a directory in a music folder e.g. say Artwork, trying to play the first track actually plays the second track, the second track plays the third track etc

**Comment by ludouzi** [comment]
Thanks, that should now be fixed. 

The browser also doesn't currently respond to underlying file system events e.g. deleting or renaming a file, so I'll be working on that next.

**Comment by ludouzi** [comment]
Closing as all requested features have been implemented.
